src/matter/cluster/OnOffCluster.ts

```diff
@@ -4,18 +4,109 @@
+ */
 export const OnOffCluster = Cluster(
     0x06,
     "On/Off",
     {
-        on: Attribute(0, BooleanT),
+        onOff: Attribute(0, BooleanT, false), /* reportable: true */
+        globalSceneControl: OptionalAttribute(0x4000, BooleanT, true),
+        onTime: OptionalWritableAttribute(0x4001, UnsignedIntT, 0),
+        offWaitTime: OptionalWritableAttribute(0x4002, UnsignedIntT, 0),
+        startUpOnOff: OptionalWritableAttribute(0x4003, OnOffStartUpOnOffT), /* nullable: true, writeAcl: manage */
     },
     {
+        /**
+         * On receipt of this command, a device SHALL enter its ‘Off’ state. This state is device dependent, but it is
+         * recommended that it is used for power off or similar functions. On receipt of the Off command, the OnTime
+         * attribute SHALL be set to 0.
+         */
         off: Command(0, NoArgumentsT, 0, NoResponseT),
+        /**
+         * On receipt of this command, a device SHALL enter its ‘On’ state. This state is device dependent, but it is
+         * recommended that it is used for power on or similar functions. On receipt of the On command, if the value
+         * of the OnTime attribute is equal to 0, the device SHALL set the OffWaitTime attribute to 0.
+         */
         on: Command(1, NoArgumentsT, 1, NoResponseT),
+        /**
+         * On receipt of this command, if a device is in its ‘Off’ state it SHALL enter its ‘On’ state. Otherwise,
+         * if it is in its ‘On’ state it SHALL enter its ‘Off’ state. On receipt of the Toggle command, if the value
+         * of the OnOff attribute is equal to FALSE and if the value of the OnTime attribute is equal to 0, the device
+         * SHALL set the OffWaitTime attribute to 0. If the value of the OnOff attribute is equal to TRUE, the OnTime
+         * attribute SHALL be set to 0.
+         */
         toggle: Command(2, NoArgumentsT, 2, NoResponseT),
+        /**
+         * The OffWithEffect command allows devices to be turned off using enhanced ways of fading.
+         */
+        offWithEffect: Command(0x40, OffWithEffectRequestT, 0x40, NoResponseT), /* optional: true */
+        /**
+         * The OnWithRecallGlobalScene command allows the recall of the settings when the device was turned off.
+         */
+        onWithRecallGlobalScene: Command(0x41, NoArgumentsT, 0x41, NoResponseT), /* optional: true */
+        /**
+         * The OnWithTimedOff command allows devices to be turned on for a specific duration with a guarded off
+         * duration so that SHOULD the device be subsequently switched off, further OnWithTimedOff commands, received
+         * during this time, are prevented from turning the devices back on.
+         */
+        onWithTimedOff: Command(0x42, OnWithTimedOffRequestT, 0x42, NoResponseT), /* optional: true */
     },
 );
+
+/*
+Features:
+  <bitmap name="OnOffFeature" type="BITMAP32">
+    <cluster code="0x0006" />
+    <field name="Lighting" mask="0x01" />
```

Comment:
In fact this is the "This cluster SHALL support the FeatureMap global attribute." topic ... so a cluster server instace should have some global Fields (ssee 7.13 in core specs)

Reply:
In fact I would see this as another object with the  "global features" when defining the cluster. I assume that "device level" will not be enough (Imagine a device that has a socket and also a light ... so you will have two onOff clusters maybe with different "global features". And yes I assume we end up in such stuff (when you see Homekit devices and such then this will allll be mixed later on)

Reply:
PS: And in fact we will need that because there are things for onoff that are only relevant when Lighing and not when no lighing ... so pot. there will be logic differences based on global features,